### PR TITLE
KEYCLOAK-14653 Changed operation type to UPDATE when updating client scopes

### PR DESCRIPTION
**Pull request by eedzjee**
Changed operation type to UPDATE when updating client scopes and added representation in the admin event.

Fix for issue: https://issues.redhat.com/browse/KEYCLOAK-14653

### CODE REVIEW

**Comment by abstractj** [comment]
@eedzjee would you like to move forward with this change? If yes, please iterate over the comments on https://issues.redhat.com/browse/KEYCLOAK-14653 and reopen the issue if you disagree.

**Comment by eedzjee** [comment]
@abstractj yes, I still think this update is needed. I added a comment in the ticket as I am able to reproduce it and the code speaks for itself.
If you want to monitor on when clients are created or deleted, the current events are misleading.

File: services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java [hunk 1]
```diff
@@ -344,7 +344,8 @@ private void addDefaultClientScope(String clientScopeId, boolean defaultScope) {
         }
         client.addClientScope(clientScope, defaultScope);
 
-        adminEvent.operation(OperationType.CREATE).resource(ResourceType.CLIENT).resourcePath(session.getContext().getUri()).success();
+        final ClientRepresentation rep = ModelToRepresentation.toRepresentation(client, session);
+        adminEvent.operation(OperationType.UPDATE).resource(ResourceType.CLIENT).resourcePath(session.getContext().getUri()).representation(rep).success();
```
Comment (hmlnarik) [inline services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java:348]:
```suggestion
                adminEvent.operation(OperationType.CREATE).resource(ResourceType.CLIENT_SCOPE_CLIENT_MAPPING).resourcePath(session.getContext().getUri()).success();
```

File: services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java [hunk 2]
```diff
@@ -360,7 +361,8 @@ public void removeDefaultClientScope(@PathParam("clientScopeId") String clientSc
         }
         client.removeClientScope(clientScope);
 
-        adminEvent.operation(OperationType.DELETE).resource(ResourceType.CLIENT).resourcePath(session.getContext().getUri()).success();
+        final ClientRepresentation rep = ModelToRepresentation.toRepresentation(client, session);
+        adminEvent.operation(OperationType.UPDATE).resource(ResourceType.CLIENT).resourcePath(session.getContext().getUri()).representation(rep).success();
```
Comment (hmlnarik) [inline services/src/main/java/org/keycloak/services/resources/admin/ClientResource.java:365]:
```suggestion
        adminEvent.operation(OperationType.DELETE).resource(ResourceType.CLIENT_SCOPE_CLIENT_MAPPING).resourcePath(session.getContext().getUri()).success();	```

**Review by hmlnarik — CHANGES_REQUESTED** [review]
Thank you for the PR. I agree that the current behaviour is wrong. The fix should be rather logging that the client scope link to a particular client has been created / deleted, so it is needed to introduce a proper `ResourceType` that would be logged together with a proper `CREATE` / `DELETE` operation. Please see inline for suggestions. When pushing the updated version, please rebase to the current master

**Review by hmlnarik — APPROVED** [review]
Thank you for the PR!

**Pull request merged**